### PR TITLE
Fix core exceptions extending and add missing doc-blocks

### PR DESCRIPTION
**Pull request by zuk3975**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | 1.7.6.x
| Description?  | Some of exceptions in PrestaShop/Core and PrestaShop/Core/Domain where extending generic php exceptions. This PR fixes to extend core, domain or specific sub-domain exception. Also adds missing docblocks, deletes unused `src/Core/Domain/Employee/Exception/UndefinedEmployeeStatusException.php`
| Type?         | bug fix
| Category?     | BO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? |
| How to test?  | Everything should work as before, no visible changes for user.

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/13522)

### CODE REVIEW

**Comment by zuk3975** [comment]
@matks ready :)

File: src/Core/Backup/Exception/BackupException.php [hunk 1]
```diff
@@ -26,11 +26,11 @@
...
 namespace PrestaShop\PrestaShop\Core\Backup\Exception;
 
-use Exception;
+use PrestaShop\PrestaShop\Core\Exception\CoreException;
 
 /**
  * Class BackupException is thrown whenever backup fails.
  */
-class BackupException extends Exception implements ExceptionInterface
+class BackupException extends CoreException implements ExceptionInterface
```
Comment (matks) [inline src/Core/Backup/Exception/BackupException.php]: We have a backup `ExceptionInterface` ? 😄

**Review by matks — DISMISSED** [review]

**Comment by matks** [comment]
I think the change is too broad, but automated tests should be enough to cover it, isn't it ?

File: src/Core/Backup/Exception/BackupException.php (hunk 1 above)
Comment (Quetzacoalt91) [reply src/Core/Backup/Exception/BackupException.php]: Isn't it a little overkill to have `CoreException` and `ExceptionInterface`, which are both empty?

Comment (matks) [reply src/Core/Backup/Exception/BackupException.php]: Yes I was surprised too. I think `ExceptionInterface` is totally useless.

Comment (Quetzacoalt91) [reply src/Core/Backup/Exception/BackupException.php]: I let you decide if this should be removed, or if we continue as-is.

Comment (zuk3975) [reply src/Core/Backup/Exception/BackupException.php]: @matks, @Quetzacoalt91 I removed it, as it seems it was used only in one place, so it should be ok.

**Review by Quetzacoalt91 — APPROVED** [review]

File: src/Core/Backup/Exception/BackupException.php (hunk 1 above)
Comment (Quetzacoalt91) [reply src/Core/Backup/Exception/BackupException.php]: Thank you @zuk3975

**Comment by matks** [comment]
Thank you @zuk3975

**Pull request merged**